sources/api/schnauzer/tests/multicall_integration.rs

```diff
@@ -0,0 +1,230 @@
+        let link_path = self.temp_dir.path().join(name);
+        symlink(self.binary_path, &link_path).expect(&format!("Failed to create {} symlink", name));
+        link_path
+    }
+
+    /// Create a copy of the binary with the given name
+    fn create_copy(&self, name: &str) -> PathBuf {
+        let copy_path = self.temp_dir.path().join(name);
+        fs::copy(self.binary_path, &copy_path)
+            .expect(&format!("Failed to copy binary to {}", name));
+
+        #[cfg(unix)]
+        {
+            use std::os::unix::fs::PermissionsExt;
+            let mut perms = fs::metadata(&copy_path)
+                .expect("Failed to get metadata")
+                .permissions();
+            perms.set_mode(0o755);
+            fs::set_permissions(&copy_path, perms).expect("Failed to set permissions");
+        }
+
+        copy_path
+    }
+
+    /// Run a command with the given binary path and arguments
+    fn run_command(&self, binary_path: &Path, args: &[&str]) -> Output {
+        Command::new(binary_path)
+            .args(args)
+            .output()
+            .expect(&format!(
+                "Failed to execute {:?} with args {:?}",
+                binary_path, args
+            ))
+    }
+
+    /// Check if the output contains v1-specific usage message
+    fn is_v1_output(&self, output: &Output) -> bool {
+        let stderr = String::from_utf8_lossy(&output.stderr);
+        stderr.contains("Usage:") && stderr.contains("SETTING_KEY")
+    }
+
+    /// Check if the output is from v2
+    fn is_v2_output(&self, output: &Output) -> bool {
+        !self.is_v1_output(output)
+    }
+}
+
+/// Test that the multicall binary can be invoked directly as schnauzer
+#[test]
+fn test_direct_schnauzer_invocation() {
+    // Given A multicall binary built by cargo
+    let test = MulticallTest::new();
+
+    // When Invoking the binary directly with --help
+    let output = test.run_command(Path::new(test.binary_path), &["--help"]);
+
+    // Then Should exit with usage error code for v1 behavior
+    assert_eq!(output.status.code(), Some(USAGE_ERROR_EXIT_CODE));
+
+    // Should contain usage information
+    let stderr = String::from_utf8_lossy(&output.stderr);
+    assert!(stderr.contains("Usage:"));
+}
+
+/// Test that the multicall binary can be invoked directly as schnauzer-v2
+#[test]
+fn test_direct_schnauzer_v2_invocation() {
+    // Given A multicall binary copied with schnauzer-v2 name
+    let test = MulticallTest::new();
+    let v2_binary_path = test.create_copy("schnauzer-v2");
+
+    // When Invoking the renamed binary with --help
+    let output = test.run_command(&v2_binary_path, &["--help"]);
+
+    // Then v2 should handle --help differently than v1
+    assert!(output.status.success() || output.status.code() == Some(2));
+    assert!(test.is_v2_output(&output));
+}
+
+/// Test symlink behavior - the core multicall functionality
+#[test]
+fn test_symlink_dispatch() {
```

Comment:
AFAICT this is the only useful integration test, the others are just repeating this.